client/src/documentLinkProvider.ts

```diff
@@ -27,20 +27,49 @@ export class BitbakeDocumentLinkProvider implements vscode.DocumentLinkProvider
       * └── files
       *     └── syslog-startup.conf
     */
+
     const filenames = linksData.map(link => link.value.split(';')[0])
     const filenamesRegex = '{' + filenames.join(',') + '}'
-    const parentDir = document.uri.path.split('/').slice(0, -1).join('/')
-    const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+    const parentDir = uri.path.split('/').slice(0, -1).join('/')
+    const pnDir = path.join(parentDir, extractRecipeName(uri.path) as string)
     const filesDir = path.join(parentDir, 'files')
-    return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
-      ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+    try {
+      return [...(await vscode.workspace.findFiles(new vscode.RelativePattern(pnDir, '**/' + filenamesRegex), undefined, filenames.length, token)),
+        ...(await vscode.workspace.findFiles(new vscode.RelativePattern(filesDir, '**/' + filenamesRegex), undefined, filenames.length, token))]
+    } catch (err) {
+      logger.error(`An error occurred when finding files with pattern: ${filenamesRegex}. ${JSON.stringify(err)}`)
+      return []
+    }
+  }
+
+  private findFolders (links: Awaited<RequestResult['getLinksInDocument']>): Awaited<RequestResult['getLinksInDocument']> {
+    return links.filter(link => path.parse(link.value).ext === '')
   }
 
   async provideDocumentLinks (document: vscode.TextDocument, token: vscode.CancellationToken): Promise<vscode.DocumentLink[]> {
     const linksData = await this.client.sendRequest<RequestResult['getLinksInDocument']>(RequestMethod.getLinksInDocument, { documentUri: document.uri.toString() })
     const documentLinks: vscode.DocumentLink[] = []
-    const relatedFiles = await this.findRelatedFiles(document, linksData, token)
-    logger.debug(`Found ${relatedFiles.length} local SRC_URI files for ${document.uri.toString()}`)
+    const relatedFiles = await this.findRelatedFiles(document.uri, linksData, token)
+    const folders = this.findFolders(linksData)
+
+    logger.debug(`Found ${folders.length} folders from the links for file: ${document.uri.toString()}`)
+    logger.debug(`Found ${relatedFiles.length} files from the links for file: ${document.uri.toString()}`)
+
+    // Reveal folders in explorer
+    folders.forEach((folder) => {
+      const parentDir = document.uri.path.split('/').slice(0, -1).join('/')
+      const pnDir = path.join(parentDir, extractRecipeName(document.uri.path) as string)
+      const commandArguments = [path.join(pnDir, folder.value)]
+      const targetUri = vscode.Uri.parse(`command:bitbake.revealInExplorer?${encodeURIComponent(JSON.stringify(commandArguments))}`)
```

Comment:
You should avoid defining a new command and use `vscode.openFolder` or `vscode.open`, any command that does the reveal properly. You should directly call `revealInExplorer`.

Reply:
yes, will commit it with the changes on finding folders.

Reply:
check 3497a08fd24717bf55e477a71a90c0afbc77968b